contrib/node/src/python/pants/contrib/node/targets/node_thrift_library.py

```diff
@@ -0,0 +1,7 @@
+from pants.contrib.node.targets.node_module import NodeModule
+
+
+class NodeThriftLibrary(NodeModule):
```

Comment:
It may be a good idea here to prevent users from putting in their own `package.json` / `yarn.lock` file here as there are quite a few areas where the user can really mess things up. Relying on the generated package.json and yarn.lock file is probably for the best.

i.e.
Check if these files exist in the sources payload, and if they do, throw an error.